File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int sum=0;
        double avg=0;
        int l=0;
        int r=k-1;
        int n=nums.length;
        for(int i=0;i<k;i++){
            sum+=nums[i];
        }
        int maxsum=sum;
        while(r<n-1){
            sum=sum-nums[l];
            l++;
            r++;
            sum=sum+nums[r];
           maxsum=Math.max(maxsum,sum);

        }
                    avg=((double) maxsum)/k;
        return avg;
        
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        sum=0
        maxsum=0
        i=0
        j=len(nums)-1
        m=k
        for r in range(0,k):
            sum+=nums[r]
        maxsum=sum
        while(m<=j):
            sum-=nums[i]
            sum+=nums[m]
            i+=1
            m+=1
            maxsum=max(maxsum,sum)
        avg=maxsum/k
        return avg

        